src/transaction/log_page_buffer.c

```diff
@@ -7303,6 +7324,81 @@ logpb_checkpoint (THREAD_ENTRY * thread_p)
 #undef detailed_er_log
 }
 
+/* log_checkpoint_trantable - TODO
+ */
+int
+logpb_checkpoint_trantable (THREAD_ENTRY * const thread_p)
+{
+  const bool detailed_logging = prm_get_bool_value (PRM_ID_LOG_CHKPT_DETAILED);
+
+  if (!is_tran_server_with_remote_storage ())
+    {
+      er_log_debug (ARG_FILE_LINE,
+		    "checkpointing trantable is only allowed on transaction server with remote storage\n");
+      return ER_FAILED;
+    }
+
+  if (detailed_logging)
+    {
+      _er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: started\n");
+    }
+
+  {
+    LOG_CS_ENTER (thread_p);
+    // *INDENT-OFF*
+    scope_exit <std::function<void (void)>> unlock_log_cs_on_exit ([thread_p] ()
+    {
+      LOG_CS_EXIT(thread_p);
+    });
+    // *INDENT-ON*
+
+    if (detailed_logging)
+      {
+	_er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: load trantable\n");
+      }
+    cublog::checkpoint_info trantable_checkpoint_info;
+
+    LOG_LSA dummy_smallest_tran_lsa = NULL_LSA;
+    trantable_checkpoint_info.load_trantable_snapshot (thread_p, dummy_smallest_tran_lsa);
+
+    const log_lsa trantable_checkpoint_lsa = trantable_checkpoint_info.get_snapshot_lsa ();
+
+    if (detailed_logging)
+      {
+	_er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: add\n");
+      }
+    log_Gl.m_metainfo.add_checkpoint_info (trantable_checkpoint_lsa, std::move (trantable_checkpoint_info));
+
+    // make sure new checkpoint is persisted to disk
+    log_write_metalog_to_file ();
+
+    logpb_flush_pages_direct (thread_p);
```

Comment:
logpb_flush_pages_direct sends the log to page server and serializes the log records into pages. The necessary condition here is to have the log safely saved until at least the snapshot LSA on the page server. To satisfy this condition I think the next condition is better:
```suggestion
    logpb_flush_pages (thread_p, trantable_checkpoint_lsa);
```